Sources/MoEngage-Swift-Segment/MoEngage_Swift_Segment.swift

```diff
@@ -0,0 +1,240 @@
+
+        guard let tempSettings: MoEngageSettings = settings.integrationSettings(forPlugin: self) else { return }
+        moengageSettings = tempSettings
+        
+        guard let appID = moengageSettings?.apiKey, let moeAppID = MoEngageInitializer.shared.config?.moeAppID else {
+            return
+        }
+        if appID != moeAppID {
+            return
+        }
+                
+        MoEngageCoreIntegrator.sharedInstance.enableSDKForSegment(instanceID: appID)
+        
+        DispatchQueue.main.async {
+            if let segmentAnonymousID = self.analytics?.anonymousId {
+                MoEngageSDKAnalytics.sharedInstance.setUserAttribute(segmentAnonymousID, withAttributeName: self.segmentAnonymousIDAttribute, forAppID: appID)
+            }
+        }
+        
+        if UNUserNotificationCenter.current().delegate == nil {
+            UNUserNotificationCenter.current().delegate = self
+        }
+    }
+    
+    public func identify(event: IdentifyEvent) -> IdentifyEvent? {
+        if let userId = event.userId, !userId.isEmpty {
+            MoEngageSDKAnalytics.sharedInstance.setUniqueID(userId, forAppID: moengageSettings?.apiKey)
+        }
+        
+        if let traits = event.traits?.dictionaryValue {
+            if let birthday = traits[UserAttributes.birthday.rawValue] as? String {
+                let dateformatter = DateFormatter()
+                dateformatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
+                if let formattedBirthday = dateformatter.date(from: birthday) {
+                    MoEngageSDKAnalytics.sharedInstance.setDateOfBirth(formattedBirthday, forAppID: moengageSettings?.apiKey)
+                }
+            }
+            
+            if let name = traits[UserAttributes.name.rawValue] as? String {
+                MoEngageSDKAnalytics.sharedInstance.setName(name, forAppID: moengageSettings?.apiKey)
+            }
+            if let birthday = traits[UserAttributes.isoBirthday.rawValue] as? String {
+                MoEngageSDKAnalytics.sharedInstance.setDateOfBirthInISO(birthday, forAppID: moengageSettings?.apiKey)
+                
+            }
+            
+            if let email = traits[UserAttributes.email.rawValue] as? String {
+                MoEngageSDKAnalytics.sharedInstance.setEmailID(email, forAppID: moengageSettings?.apiKey)
+            }
+
+            if let firstName = traits[UserAttributes.firstName.rawValue] as? String {
+                MoEngageSDKAnalytics.sharedInstance.setFirstName(firstName, forAppID: moengageSettings?.apiKey)
+            }
+
+            if let lastName = traits[UserAttributes.lastName.rawValue] as? String {
+                MoEngageSDKAnalytics.sharedInstance.setLastName(lastName, forAppID: moengageSettings?.apiKey)
+            }
+            
+            if let gender = traits[UserAttributes.gender.rawValue] as? String {
+                if gender.lowercased() == "m" || gender.lowercased() == "male" {
```

Comment:
@deepa-iOS-SDK lets use gender =  gender.lowercased() and use it instead of converting to lower case multiple times